Document new Redirects feature (#3395)
withastro/docs: src/content/docs/en/core-concepts/routing.mdx

</html>
```

> ## Redirects (experimental)
> 

Reviewer: ```suggestion
## Redirects

Sometimes you will need to redirect your readers to a new page, either permanently because your site structure has changed, or in response to an action such as logging in to an authrenticated route.

You can define rules to [redirect users to permanently-moved pages](#configured-redirects-experimental) in your Astro config. Or, [redirect users dynamically](#dynamic-redirects) as they use your site.

### Configured Redirects (experimental)
```
Especially since only the new (permanent, configured) redirects are actually experimental, I think for now this sectioning makes sense? Both types of redirects are sub-headings of redirects?

Author: Yeah, good point, I like this.

<Since v="2.6.0" />

You can specify a mapping of redirects in your Astro config. For most redirects, this is a mapping of old route to the new route:

```js
import { defineConfig } from 'astro/config';

export default defineConfig({
  experimental: {
    redirects: false
  },
  redirects: {
    '/old-page': '/new-page'
  }
});
```

These redirects follow the same rules as file-based routes. Dynamic routes are allowed as long as both the new and old route contain the same parameters, for ex:

```js
{
  "/blog/[...slug]": "/articles/[...slug]"
}
```

> You can also specify the status code by providing an object as the value:

Reviewer: ```suggestion
Using SSR or a static adapter, you can also provide an object as the value, allowing you to specify the `status` code in addition to the new `destination`:
```

</html>
```

## Redirects

> Sometimes you will need to redirect your readers to a new page, either permanently because your site structure has changed, or in response to an action such as logging in to an authenticated route.

Reviewer 2: Unnecessary comma
```suggestion
Sometimes you will need to redirect your readers to a new page, either permanently because your site structure has changed or in response to an action such as logging in to an authenticated route.
```

> Sometimes you will need to redirect your readers to a new page, either permanently because your site structure has changed, or in response to an action such as logging in to an authrenticated route.

Reviewer: ```suggestion
Sometimes you will need to redirect your readers to a new page, either permanently because your site structure has changed, or in response to an action such as logging in to an authenticated route.
```

You can define rules to [redirect users to permanently-moved pages](#configured-redirects-experimental) in your Astro config. Or, [redirect users dynamically](#dynamic-redirects) as they use your site.

### Configured Redirects (experimental)
<Since v="2.6.0" />

After enabling the experimental flag `redirects: true`, you can specify a mapping of permanent redirects in your Astro config. For most redirects, this is a mapping of an old route to the new route:

```js title="astro.config.mjs" {5, 7-9}
import { defineConfig } from 'astro/config';

export default defineConfig({
  experimental: {
    redirects: true
  },
  redirects: {
    '/old-page': '/new-page'
  }
});
```

> These redirects follow the same rules as file-based routes. Dynamic routes are allowed as long as both the new and old route contain the same parameters, for example:

Reviewer 2: Not sure how this works in English, but "new and old" is an implicit plural, and supposedly "route" should be changed to agree in number.
```suggestion
These redirects follow the same rules as file-based routes. Dynamic routes are allowed as long as both the new and old routes contain the same parameters, for example:
```

Reviewer: yes, this absolutely correct, Yan!

```js
{
  "/blog/[...slug]": "/articles/[...slug]"
}
```

Using SSR or a static adapter, you can also provide an object as the value, allowing you to specify the `status` code in addition to the new `destination`:

```js
import { defineConfig } from 'astro/config';

export default defineConfig({
  experimental: {
    redirects: true
  },
  redirects: {
    '/old-page': {
      status: 302,
      destination: '/new-page'
    }
  }
});
```

> When running `astro build` Astro will write out HTML files with the [meta refresh](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#examples) tag by default. Supported adapters will instead write out the host's configuration file with the redirects.

Reviewer: ```suggestion
When running `astro build`, Astro will output HTML files with the [meta refresh](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta#examples) tag by default. Supported adapters will instead write out the host's configuration file with the redirects.
```
